optimum/amd/brevitas/accelerate_utils.py

```diff
@@ -331,7 +331,11 @@ def remove_hooks(model: torch.nn.Module):
 def update_internal_dict(module):
     prefix = module._hf_hook.weights_map.prefix
     for key in module.state_dict().keys():
-        module._hf_hook.weights_map.dataset.state_dict[prefix + key] = recurse_getattr(module, key + ".data")
+        # It might happen that we call an quantization's inner modules, and this cause some parameters to be
+        # already on meta device. This is not a problem for their value but we need to check here
+        curr_device = (recurse_getattr(module, key + ".data")).device
+        if str(curr_device) != 'meta':
+            module._hf_hook.weights_map.dataset.state_dict[prefix + key] = (recurse_getattr(module, key + ".data")).cpu()
```

Comment:
Why may some parameters be on meta device, and why is the conditional write in `state_dict` needed?

Reply:
The execution flow is something like that:
assuming that we have a module with a single submodule.

```
some func:
    call allocate_params:
          call pre_forward module
              call pre_forward submodule
      
     call forward on submodule (that triggers pre_forward, _old_forward and post_forward) <- weights are moved on 
         meta device

     call offload_params:
              update_dict on module and submodule's parameters <- need to check if weights are already on meta device
              call post_forward on module
              call post_forward on submodule <- here it would break as well for the reason mentioned above 
```